### DISCUSSION TITLE
Accessing selected revision from a plugin

### DISCUSSION THREAD

**Discussion by ahmeturun** [gitextensions/gitextensions#8560] [General] [answered]
Hi everyone, I'm trying to create a gitlab plugin. in the plugin I open a form to create a merge request, and while opening the form I need to have the selected revision from the revisin control to be selected in a dropdown of my form. I've tried to access the "[FormBrowse](https://github.com/gitextensions/gitextensions/blob/master/GitUI/CommandsDialogs/FormBrowse.cs)" which has the [Revision grid control](https://github.com/gitextensions/gitextensions/blob/master/GitUI/CommandsDialogs/FormBrowse.cs#L116) on it, but this gives me error on runtime 'System.Collections.Generic.IReadOnlyList1<GitCommands.GitRevision> GitUI.RevisionGridControl.GetSelectedRevisions(System.Nullable1<GitUI.SortDirection>)".
here is the snippet for what I explained above:

```c#
	[Export(typeof(IGitPlugin))]
	public class GitLabCreateMergeRequest : GitPluginBase
	{
		public static string GitLabCreateMRDescription = "gitlabcreatemergerequest";
		public GitLabCreateMergeRequest() : base(false)
		{
			SetNameAndDescription(GitLabCreateMRDescription);
		}
		public override bool Execute(GitUIEventArgs args)
		{
			System.Diagnostics.Debugger.Launch();
			if (!TryGetRepositoryHost(args.GitModule, args.OwnerForm, out var repoHostPlugin))
			{
				return false;
			}
			var ss = (args.OwnerForm as GitModuleForm).RevisionGridControl;
                       // runtime exception on this line
			ss.GetCurrentRevision();
                }
        }
```
does anyone can tell me if there is a way to do what I'm trying here, thanks.

**Comment by ahmeturun** [comment, marked as answer]
ok so I found out what the problem was. I was copying the plugin dlls in to my locally installed GitExtension plugins path for testing and the refrenced GitExtensions dlls from my plugin were loading the one on the development folder. I made sure that the dlls were matching with the running GitExtensions dlls and everything works fine now, thanks.

**Reply by RussKie** [reply]
This _may_ result in a NRE:
```cs
var ss = (args.OwnerForm as GitModuleForm).RevisionGridControl;
```

And you'd want to be defensive here, and check for null.

**Reply by ahmeturun** [reply]
you're right, I'll add a null check there, thanks.
